scout/adapter/mongo/case.py

```diff
@@ -904,7 +905,10 @@ def load_case(self, config_data: dict, update: bool = False, keep_actions: bool
                     continue
                 load_variants.add((vcf_file["variant_type"], vcf_file["category"]))
 
-            for variant_type, category in load_variants:
+            for variant_type, category in sorted(
```

Comment:
Grasping of the lambda function for sorting has never been my forte. Having admitted this, I find it quite complicated to follow when also calling the new `get_lowest_load_priority` function. 
Also the naming **lowest load priority** is confusing to me when I would expect that the **high** prio ones would be first. 

I was wondering if the function is avoidable at all.. Like if at line 887 of this file you have:

```
# load from files
        files = [
            {
                "file_name": file_type,
                "variant_type": FILE_TYPE_MAP[file_type]["variant_type"],
                "category": FILE_TYPE_MAP[file_type]["category"],
                "priority": FILE_TYPE_MAP[file_type]["load_priority"] <--------------------------
            }
            for file_type in FILE_TYPE_MAP.keys()
            if FILE_TYPE_MAP[file_type]["variant_type"] != "research"
        ]
```
And then again at line 905:
`load_variants.add((vcf_file["variant_type"], vcf_file["category"], vcf_file["priority"]))`


And then more simply (NOT simply! 😆 )

`sorted(load_variants, key=lambda tup: tup[2])` ??

But perhaps I am missing something. I haven't spent as much time on it as you did..

Reply:
It would be really simple if we only ever wanted to sort on category prio. The tricky thing is that it’s two different calls, an outer one asking to load the highest prio category, type and then an inner one asking for the highest prio file. If the call was in one loop like directly enumerating the files it would also be very simple.

Reply:
And lowest prio value first is standard in *nix isn’t it? As in lowest nice value goes first. It is also closest to an order numbering without having to make it contiguous.

Reply:
> lowest prio value first is standard in *nix isn’t it?

We are human beings before being programmers, aren't we? ;)

Reply:
And humans never say prio 1?

Reply:
Whatever, glad midsommar! :)

Reply:
This appears a classic design conflict, with no obvious resolution. Lower numer, higher prio is more common (defcon, unix system prio, and as spoken), but it is by no means completely obvious. I added some more explanation on this to the docstrings.